### ISSUE TITLE
Disable upload button

### ISSUE THREAD

**Issue by kwon14** [jupyterlab/jupyterlab#10405]
Hello.!
Thank you very much for your hard work.

Disable the upload feature as a usage policy.
As a result of browsing several websites
jupyterlab/packages/filebrowser/src/browser.ts
I've seen that you can just delete the "this.toolbar.addItem('upload', uploader);" line of code.

the question is
1. Is there a way to find out what the browser.ts file is?
2. Is there a way to block the upload button or function?

As always, thank you for your reply
Thank yo
![스크린샷 2021-06-14 오후 4 43 17](https://user-images.githubusercontent.com/64061288/121857358-80ebe680-cd30-11eb-97db-7fd65179b2bb.png)
u.!!

**Comment by jtpio** [comment]
Thanks @kwon14.

Maybe this functionality could then be moved to its own plugin so it can be disabled with the `jupyter labextension disable` command?

Similar to the download UI and feature: https://github.com/jupyterlab/jupyterlab/pull/10065

**Comment by jtpio** [comment]
Somewhat related: https://github.com/jupyterlab/jupyterlab/issues/7451

**Comment by Jose-Matsuda** [comment]
Hello @jtpio, I've been going through issues in this and the notebook repo looking for more information on blocking the upload of files via drag and drop into jupyter but haven't found anything solid in terms of solutions or next steps. Would you happen to have any suggestions for that? 

Something we tried was blocking the PUT network call to the path of notebook/notebookName/api/contents/ (via an [Istio Authorization policy](https://istio.io/latest/docs/reference/config/security/authorization-policy/)) but as it turns out that the PUT to that path is the same that is used when jupyter saves a file so that wouldn't work.

**Comment by jtpio** [comment]
@Jose-Matsuda right, the drag and drop upload is implemented in the file browser directly, so it's not as easy as disabling a core JupyterLab plugin unfortunately.

https://github.com/jupyterlab/jupyterlab/pull/17676 explores adding a new option to disable file uploads via drag and drop, although it would also need a way to prevent users from going to the settings editor and changing that option.

Otherwise for the original issue of disabling the upload button, it should be possible to put something like the following in an `overrides.json` file:

```json
{
  "@jupyterlab/filebrowser-extension:widget": {
    "toolbar": [
      {
        "name": "uploader",
        "disabled": true
      }
    ]
  }
}
```

**Comment by MMesch** [comment]
The triaging agent has identified this issue as likely resolved (Rationale below). @jtpio can you confirm and close or add what is still missing otherwise?

---

**Agent Rationale** : This issue has been fully resolved through the implementation of the allowFileUploads setting in JupyterLab. The solution comprehensively addresses the original request by:

- Upload Button Control: The Uploader widget respects the allowFileUploads setting and disables the button when set to false
- Drag-and-Drop Prevention: DirListing component prevents drag-and-drop uploads when disabled
- Model-Level Protection: FileBrowserModel throws an error if upload is attempted when disabled
- Configuration Interface: Setting is exposed in the browser.json schema for easy configuration

The implementation was completed in commit b5b497ba on 2025-08-08 by Jeremy Tuloup and provides exactly what was requested - a way to disable file uploads for policy compliance. The solution is more robust than the original suggestion of just removing the toolbar button, as it prevents uploads through all vectors.

**Comment by jtpio** [comment]
Opened https://github.com/jupyterlab/jupyterlab/pull/18040 to document how to disable the upload button, which will help close that issue.

As mentioned above there is more general issue about file uploads, so folks interested in this topic can continue the conversation in https://github.com/jupyterlab/jupyterlab/issues/7451.
